Origin: vendor,  https://github.com/MariaDB/server/commit/df4fddb8961355fc68f32375d2bac5735057f205.patch
Subject: [PATCH] MDEV-10428 main.information_schema_stats fails sporadically
 in buildbot

Resultsets which contain more than one row need to be sorted
---
 mysql-test/r/information_schema_stats.result | 4 ++--
 mysql-test/t/information_schema_stats.test   | 2 ++
 2 files changed, 4 insertions(+), 2 deletions(-)

--- a/mysql-test/r/information_schema_stats.result
+++ b/mysql-test/r/information_schema_stats.result
@@ -16,8 +16,8 @@ count(*)
 2
 select * from information_schema.index_statistics where table_schema='test' and table_name='just_a_test';
 TABLE_SCHEMA	TABLE_NAME	INDEX_NAME	ROWS_READ
-test	just_a_test	IND_just_a_test_state	2
 test	just_a_test	IND_just_a_test_first_name_last_name	1
+test	just_a_test	IND_just_a_test_state	2
 select * from information_schema.table_statistics where table_schema='test' and table_name='just_a_test';
 TABLE_SCHEMA	TABLE_NAME	ROWS_READ	ROWS_CHANGED	ROWS_CHANGED_X_INDEXES
 test	just_a_test	18	5	5
@@ -56,9 +56,9 @@ count(*)
 3
 select * from information_schema.index_statistics where table_schema='test' and table_name='just_a_test';
 TABLE_SCHEMA	TABLE_NAME	INDEX_NAME	ROWS_READ
+test	just_a_test	PRIMARY	5
 test	just_a_test	first_name	1
 test	just_a_test	state	2
-test	just_a_test	PRIMARY	5
 select * from information_schema.table_statistics where table_schema='test' and table_name='just_a_test';
 TABLE_SCHEMA	TABLE_NAME	ROWS_READ	ROWS_CHANGED	ROWS_CHANGED_X_INDEXES
 test	just_a_test	8	5	15
--- a/mysql-test/t/information_schema_stats.test
+++ b/mysql-test/t/information_schema_stats.test
@@ -13,6 +13,7 @@ alter table just_a_test add key IND_just
 alter table just_a_test add key IND_just_a_test_state(state);
 select count(*) from just_a_test where first_name='fc' and last_name='lc';
 select count(*) from just_a_test where state = 'California';
+--sorted_result
 select * from information_schema.index_statistics where table_schema='test' and table_name='just_a_test';
 select * from information_schema.table_statistics where table_schema='test' and table_name='just_a_test';
 alter table just_a_test drop key IND_just_a_test_first_name_last_name;
@@ -36,6 +37,7 @@ insert into just_a_test values(1,'fa','l
 select count(*) from just_a_test where first_name='fc' and last_name='lc';
 select count(*) from just_a_test where state = 'California';
 select count(*) from just_a_test where id between 2 and 4;
+--sorted_result
 select * from information_schema.index_statistics where table_schema='test' and table_name='just_a_test';
 select * from information_schema.table_statistics where table_schema='test' and table_name='just_a_test';
 drop table just_a_test;
